### PR TITLE
Vendoring, root go.mod and Dockerfile updates

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This does the following
- Have one `go.mod` on the root
- Vendor dependencies
- Move the Dockerfiles to another place
- Update tektoncd dependencies 😅 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [x] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: images/api.Dockerfile [hunk 1]
```diff
@@ -14,10 +14,10 @@ WORKDIR /app
 
 COPY --from=builder /go/src/github.com/tektoncd/hub/api/api-server /app/api-server
 
-# For each new version, doc has to be copied 
+# For each new version, doc has to be copied
 COPY gen/http/openapi3.json /app/docs/openapi3.json
```
Comment (sm43) [inline images/api.Dockerfile:18]: COPY gen/http/openapi3.json /app/docs/openapi3.json
```suggestion
COPY api/gen/http/openapi3.json /app/docs/openapi3.json
```

File: images/api.Dockerfile [hunk 2]
```diff
@@ -14,10 +14,10 @@ WORKDIR /app
...
 COPY v1/gen/http/openapi3.json /app/docs/v1/openapi3.json
```
Comment (sm43) [inline images/api.Dockerfile:19]: COPY api/v1/gen/http/openapi3.json /app/docs/v1/openapi3.json

**Comment by piyush-garg** [comment]
Better if you can squash the first and last commit only

**Comment by piyush-garg** [comment]
@vdemeester Can you keep go version 1.14 as like before

**Comment by vdemeester** [comment]
@piyush-garg oh, one was 1.13 and one 1.14 I see. I'll update and squash 😉

**Pull request merged**